# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1424327516 18000
#      Thu Feb 19 01:31:56 2015 -0500
# Node ID 47440b762547a67c878a43d0dfa611b381e00520
# Parent  ed1bafbb87ee7d3373a77f862c060edd0ab743e7
bring back mouse_wheel_zoom function as deprecated

* mouse_wheel_zoom.m: New function.
* scripts/deprecated/module.mk: Update.
* NEWS: Note that mouse_wheel_zoom is deprecated.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -202,16 +202,17 @@ Summary of important user-visible change
       delaunay3            | delaunay
       dump_prefs           | individual preference get/set routines
       find_dir_in_path     | dir_in_loadpath
       finite               | isfinite
       fmod                 | rem
       fnmatch              | glob or regexp
       loadaudio            | audioread
       luinc                | ilu or ichol
+      mouse_wheel_zoom     | mousewheelzoom axes property
       nfields              | numfields
       octave_tmp_file_name | tempname
       playaudio            | audioplayer
       saveaudio            | audiowrite
       syl                  | sylvester
       usage                | print_usage
 
       allow_noninteger_range_as_index
diff --git a/scripts/deprecated/module.mk b/scripts/deprecated/module.mk
--- a/scripts/deprecated/module.mk
+++ b/scripts/deprecated/module.mk
@@ -17,16 +17,17 @@ deprecated_FCN_FILES = \
   deprecated/java_debug.m \
   deprecated/java_invoke.m \
   deprecated/java_new.m \
   deprecated/java_unsigned_conversion.m \
   deprecated/javafields.m \
   deprecated/javamethods.m \
   deprecated/loadaudio.m \
   deprecated/luinc.m \
+  deprecated/mousewheelzoom.m \
   deprecated/nfields.m \
   deprecated/octave_tmp_file_name.m \
   deprecated/playaudio.m \
   deprecated/re_read_readline_init_file.m \
   deprecated/read_readline_init_file.m \
   deprecated/saveaudio.m \
   deprecated/saving_history.m \
   deprecated/setaudio.m \
diff --git a/scripts/deprecated/mouse_wheel_zoom.m b/scripts/deprecated/mouse_wheel_zoom.m
new file mode 100644
--- /dev/null
+++ b/scripts/deprecated/mouse_wheel_zoom.m
@@ -0,0 +1,55 @@
+## Copyright (C) 2007-2013 Shai Ayal
+##
+## This file is part of Octave.
+##
+## Octave is free software; you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 3 of the License, or (at
+## your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <http://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn {Loadable Function} {@var{old_val} =} mouse_wheel_zoom (@var{new_val})
+## Query or set the mouse wheel zoom factor.
+##
+## The zoom factor is a number in the range (0,1) which is the
+## percentage of the current axis limits that will be used when zooming.
+## For example, if the current x-axis limits are [0, 50] and
+## @code{mouse_wheel_zoom} is 0.4 (40%), then a zoom operation will
+## change the limits by 20.
+#### @end deftypefn
+
+## Deprecated in 4.0
+
+function retval = mouse_wheel_zoom (val)
+
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:deprecated-function",
+             "mouse_wheel_zoom is obsolete and will be removed from a future version of Octave, please use the mousehweelzoom axes property instead");
+  endif
+
+  if (nargin != 1)
+    print_usage ();
+  endif
+
+  fig = get (0, "currentfigure");
+  if (isempty (fig))
+    retval = get (0, "defaultaxesmousewheelzoom");
+    set (0, "defaultaxesmousewheelzoom", val);
+  else
+    retval = get (gca, "mousewheelzoom");
+    set (gca, "mousewheelzoom", val);
+  endif
+
+endfunction
+
